tasks/actorstate/miner/deadline_info.go

```diff
@@ -36,7 +36,9 @@ func (DeadlineInfoExtractor) Extract(ctx context.Context, a actorstate.ActorInfo
 		if err != nil {
 			return nil, err
 		}
-		if prevDeadlineInfo == currDeadlineInfo {
+		// TODO implement equality function
+		// dereference pointers to check equality
+		if *prevDeadlineInfo == *currDeadlineInfo {
```

Comment:
This is brittle. We have to be sure that `err == nil` guarantees we're protected from a nil-deref panic here and forever into the future. How is that being guaranteed? I would prefer this check is explicit here instead of expecting it inside the caller. If we have a local equality function doing that, that could work just a well.